src/SWP/Bundle/SettingsBundle/Manager/SettingsManager.php

```diff
@@ -0,0 +1,261 @@
+        $setting = $this->getSettingFromRepository($name, $scope, $owner);
+        if (null === $setting) {
+            /** @var SettingsInterface $setting */
+            $setting = $this->settingsFactory->create();
+            $setting->setName($name);
+            $setting->setScope($scope);
+            if (null !== $owner) {
+                $setting->setOwner($owner->getId());
+            }
+            $this->settingsRepository->persist($setting);
+        } else {
+            $setting->setUpdatedAt(new \DateTime());
+        }
+
+        $setting->setValue($this->encodeValue($defaultSetting['type'], $value));
+        $this->settingsRepository->flush();
+
+        return $setting;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function clear(string $name, $scope = ScopeContextInterface::SCOPE_GLOBAL, SettingsOwnerInterface $owner = null)
+    {
+        $this->validateScope($scope, $owner);
+
+        $setting = $this->getSettingFromRepository($name, $scope, $owner);
+        if (null !== $setting) {
+            $this->settingsRepository->remove($setting);
+
+            return true;
+        }
+
+        return false;
+    }
+
+    protected function validateScope($scope, $owner = null)
+    {
+        if (!in_array($scope, $this->scopeContext->getScopes())) {
+            throw new InvalidScopeException($scope);
+        }
+
+        if ($scope !== ScopeContextInterface::SCOPE_GLOBAL && null === $owner) {
+            throw new InvalidOwnerException($scope);
+        }
+    }
+
+    private function getFromConfiguration(string $scope = null, $name = null)
+    {
+        $settings = [];
+        if ($name !== null && array_key_exists($name, $this->settingsConfiguration)) {
+            $setting = $this->settingsConfiguration[$name];
+            if ($setting['scope'] === $scope || null === $scope) {
+                return $settings[$name] = $setting;
+            }
+
+            throw new InvalidScopeException($scope);
+        } elseif ($name !== null) {
+            throw new \Exception('There is no setting with this name.');
+        }
+
+        foreach ($this->settingsConfiguration as $name => $setting) {
+            if ($setting['scope'] === $scope || null === $scope) {
+                $setting['value'] = $this->decodeValue($setting['type'], $setting['value']);
+                $settings[$name] = $setting;
+            }
+        }
+
+        return $settings;
+    }
+
+    /**
+     * @return array|mixed
+     */
+    private function getSettingsFromRepository()
+    {
+        return $this->settingsRepository->findAllByScopeAndOwner($this->scopeContext)->getQuery()->getResult();
+    }
+
+    /**
+     * @param string                      $name
+     * @param string                      $scope
+     * @param SettingsOwnerInterface|null $owner
+     *
+     * @return mixed
+     */
+    private function getSettingFromRepository(string $name, string $scope, SettingsOwnerInterface $owner = null)
```

Comment:
`findOneByNameAndScopeAndOwner` is already in repository and returns query builder - here we get query and decide what kind of result we want to get.